docs/UserGuide.md

```diff
@@ -90,11 +90,70 @@ outdated systems, EffiTrack offers a centralized platform for easy and accurate
+    * Special characters are not allowed in names.
+    * Employee of the same name are not allowed to be added to EffiTrack, or edit an existing employee's name to the same name as another existing employee in EffiTrack.
+    * `NAME` is case-sensitive (e.g. `john Doe` is not the same as `John Doe`)
+    * `NAME` is space-sensitive (e.g. `John Doe` is not the same as `John      Doe`)
+
+* `PHONE_NUMBER`: The phone number of the employee.
+    * Has a minimum limit of 3 digits and maximum limit of 15 digits.
+
+* `EMAIL`: The email address of the employee.
+    * Valid as long as it follows the format of `local-part@domain and should not include whitespace
+    * `local-part` should only contain alphanumeric characters and the following special characters `+`, `_`, `.` and `-` is allowed.
+    * `local-part` may not start or end with any special characters.
+    * `local-part` must be followed with an `@`.
+    * `domain` is made up of one or more `domain label`.
+        * Each `domain label` is separated by a `.`.
+        * Last `domain label` must be at least 2 characters long.
+        * Each `domain label` must start and end with alphanumeric characters.
+        * Each `domain label` contains alphanumeric characters, separated only by `-`, if any.
+
+* `ADDRESS`: The home address of the employee.
+    * Can take any values, but should not be blank.
+  
+* `DEPARTMENT`: The department of the employee.
+    * Only alphanumeric department names with spaces is allowed.
+    * Special characters are not allowed in department names.
+
+* `EFFICIENCY`: The efficiency score of the employee.
+    * Takes in integer values in the range 0 to 100.
+
+* `TAG`: The tag of the employee.
+    * Should be alphanumeric without any whitespace. e.g: SisterBrother
+  
+* `INDEX`: The index refers to the index number shown in the displayed employee list.
+    * Assigned to employee based on the order added into employee list.
+    * Must be a **positive integer** (e.g. 1, 2, 3, …​) and should not be out of bound of the employee list.
+    * Can only take integer within the maximum bound of `int` data type (up to 2147483647), otherwise it will not be considered as a positive integer.
+
+* `TASK`: The task title of task.
+    * Only alphanumeric task titles with spaces is allowed.
+    * Special characters are not allowed in task titles.
+    * Task of the same task title are not allowed to be assigned to different employees.
+    * `TASK` is case-sensitive (e.g. `project` is not the same as `Project`)
+    * `TASK` is space-sensitive (e.g. `Project A` is not the same as `Project      A`)
+
+* `KEYWORD`: The keyword of the query.
+    * `KEYWORD` is case-insensitive.
+
+* `THRESHOLD`: The threshold of efficiency score of the employee.
+    * Takes in integer values in the range 0 to 100.
+
+* `COMMENT`: The comment about the employee.
+    * Only alphanumeric comments with spaces is allowed.
+    * Special characters are not allowed in comments.
```

Comment:
special characters are allowed in comments too

Reply:
no u used name validation regex